bullet_train/app/helpers/account/teams_helper.rb

```diff
@@ -2,7 +2,7 @@ module Account::TeamsHelper
   def current_team
     # TODO We do not want this to be based on the `current_team_id`.
     # TODO We want this to be based on the current resource being loaded.
-    @team || current_user&.current_team
+    current_user&.current_team || @team
```

Comment:
I'm wondering how wide ranging the effects of this could be, but I don't really know. Are there a few specific call sites that we need to update for this, if so what are they? I don't know what they are, but I'm curious if that might mean we could make changes that are perhaps lesser reaching.

Reply:
It's been a while since I last looked at this PR so I can't tell for sure, but the reason I prioritized `current_user&.current_team` here is for a reason similar to what I commented about `... && current_user.teams.include?(@team)` below. Basically, we were just setting whatever `@team` showed up as a resource to `current_team`, when I figured we should return the `current_user`'s current team first. Looking at this again though I'm not entirely sure what the intention of the original setup was, but if we decide to use `Current` to handle this then maybe we can rethink this portion altogether.

Reply:
Yeah, of the 3 lines changed in this PR this is the one that gives me pause. I do believe it will change some behavior, and maybe not in a way that we want.

For instance, imagine you belong to Team A & Team B. The last team that you manually changed to is Team A, so `current_user.current_team` is Team A.

A collaborator on Team B sends you a link to a project that Team B owns. Something like `/teams/team_b_id/projects/42`.

Currently/previously (before this PR) the `current_team` would be set to Team B because that's what's been loaded as `@team` in the process of retrieving the Project.

I think this PR would make it so that `current_team` is Team A, since that's the last thing that's been persisted to `current_user.current_team`. I don't think that's what we want.

I'll try to confirm or disconfirm that theory.

Reply:
> I think this PR would make it so that current_team is Team A, since that's the last thing that's been persisted to current_user.current_team. I don't think that's what we want.

Good point, @jagthedrummer!

Reply:
I pulled down this branch and did some testing, and the concern that I expressed doesn't appear to be valid.

It looks like we set `current_user.current_account` on every page load. Which seems a little surprising to me.

The reason we do so is that in the starter repo the `Account::ApplicationController` does `include Account::Controllers::Base`. And then scaffolded controllers inherit from `Account::ApplicationController`. In `Account::Controllers::Base` there's a declaration for `before_action :ensure_onboarding_is_complete_and_set_next_step`, which calls `ensure_onboarding_is_complete`. And in `ensure_onboarding_is_complete` we do:

```ruby
current_user.update(current_team: @team) if @team&.persisted? && current_user.teams.include?(@team)
```

I kind of wouldn't expect that we'd call `ensure_onboarding_is_complete` on literally every sign-in page load, but I could see how/why we do.

I think that if we're confident that calling `ensure_onboarding_is_complete` on every page load is what we _want_ to be doing, and not just a thing that we happen to be doing, then I think this might be fine. (Though messing with something as foundational as `current_team` does still give me a bit of the willies.)